vc-worker.py

```diff
@@ -198,6 +198,58 @@ def action_take_screenshot(request, machine, vc, action, conn):
     return {'machine.state': '<unchanged>'}
 
 
+def action_take_snapshot(request, machine, vc, conn):
+    if request.subject_id:
+        snap_ro = data.Snapshot.get_one({'_id': request.subject_id}, conn=conn)
+        result = vc.take_snapshot(uuid=machine.provider_id, snapshot_name=snap_ro.name)
+        snap = data.Snapshot.get_one_for_update({'_id': request.subject_id}, conn=conn)
+        snap.status = 'success' if result is True else 'failed'
+        snap.save(conn=conn)
+        if result is True:
+            # attach snapshot from machine if creating was successful
+            machine = data.Machine.get_one_for_update({'_id': machine.id}, conn=conn)
+            machine.snapshots.append(snap.id)
+            machine.save(conn=conn)
+
+    else:
+        settings.raven.captureMessage('Error obtaining subject_id from Request')
```

Comment:
What do you think, would it add some value to have the message a bit more specific like:
`'Error obtaining request subject_id for take snapshot'`